File: 240421/순서를 바꾸었을 때 같은 단어인지 판별하기/determine-same-word.py
A = list(input())
B = list(input())
A.sort()
B.sort()

is_same = True
for a, b in zip(A,B):
    if a != b:
        is_same = False
        break
if is_same:
    print("Yes")
else:
    print("No")

File: 240603/순서를 바꾸었을 때 같은 단어인지 판별하기/determine-same-word.java
import java.util.*;
import java.util.stream.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        char[] firstWord = br.readLine().toCharArray();
        char[] secondWord = br.readLine().toCharArray();

        if (firstWord.length != secondWord.length) {
            System.out.println("No");
            return;
        }

        Arrays.sort(firstWord);
        Arrays.sort(secondWord);

        for (int i=0; i<firstWord.length; i++) {
            if (firstWord[i] != secondWord[i]) {
                System.out.println("No");
                return;
            }
        }

        System.out.println("Yes");
    }
}